Source/Core/DolphinQt/Debugger/MemoryWidget.cpp

```diff
@@ -426,14 +445,15 @@ void MemoryWidget::OnSearchAddress()
 {
   bool good;
   u32 addr = m_search_address->text().toUInt(&good, 16);
+  u32 offset = m_search_offset->text().toUInt(nullptr, 16);  // Returns 0 if conversion fails
```

Comment:
Why is an invalid string in the address field an error but an invalid string in the offset field okay? This could probably be handled better.

Reply:
I omitted it simply because the offset is just a modifier and unable to supply an incorrect value to the address. If the offset input is invalid, the address is not effected. The offset is expected to be blank, so this is not an error state. If the offset is malformed, it simply does not modify the address. The address itself is checked to see if the number is true hexadecimal, but the only result of that error detection is making the address field bold and red.

I can see that being handled better by making an indication that the offset is malformed in the same way that the address is. I intend on making another commit for this debugger, so if this is merged before I make this edit I will include it in the next one (or it's own one with similar edits).

Reply:
Pushed a change to add this error handling in. Even if it's not fully what you had in mind, it should be better than before.

Reply:
Yeah this is definitely better.